lib/decompress.ps1

```diff
@@ -74,17 +80,20 @@ function Expand-7zipArchive {
     }
     $Status = Invoke-ExternalCommand $7zPath $ArgList -LogPath $LogPath
     if (!$Status) {
-        abort "Failed to extract files from $Path.`nLog file:`n  $(friendly_path $LogPath)"
+        abort "Failed to extract files from $Path.`nLog file:`n  $(friendly_path $LogPath)`n$(new_issue_msg $app $bucket 'decompress error')"
+    }
+    if (!$IsTar -and $ExtractDir) {
+        movedir "$DestinationPath\$ExtractDir" $DestinationPath | Out-Null
     }
     if (Test-Path $LogPath) {
         Remove-Item $LogPath -Force
     }
-    if ((strip_ext $Path) -match '\.tar$' -or $Path -match '\.t[abgpx]z2?$') {
+    if ($IsTar) {
         # Check for tar
         $Status = Invoke-ExternalCommand $7zPath @('l', "`"$Path`"") -LogPath $LogPath
         if ($Status) {
             $TarFile = (Get-Content -Path $LogPath)[-4] -replace '.{53}(.*)', '$1' # get inner tar file name
```

Comment:
This `[-4]` thing doesn't work for me.
```
Y:\
λ cat bla.txt

7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21

Scanning the drive for archives:
1 file, 1804230 bytes (1762 KiB)

Listing archive: .\radare2-extras-3.5.0.tar.gz

--
Path = .\radare2-extras-3.5.0.tar.gz
Type = gzip
Headers Size = 10

   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
                    .....     12390400      1804230  radare2-extras-3.5.0.tar
------------------- ----- ------------ ------------  ------------------------
                              12390400      1804230  1 files
Y:\
λ (Get-Content -Path 'bla.txt')[-4] -replace '.{53}(.*)', '$1'
------------------------
Y:\
```

Most of the tar files have a double extension, so why not use `[IO.Path]::GetFileNameWithoutExtension($Path)` instead?

Reply:
> Most of the tar files have a double extension, so why not use [IO.Path]::GetFileNameWithoutExtension($Path) instead?

This is for use cases when the nested filename is not equal to archive name. #3080

Reply:
Okay. But just using a line index isn't reliable.
In https://github.com/lukesampson/scoop/commit/079c4a57269a2a570e29d4326fcb615e98f0b0b4 it was `-3` no it's `-4` 😕

Reply:
I hoped it is static as asked in https://github.com/lukesampson/scoop/pull/3204#pullrequestreview-222929383

@niheaven Why the index changed?

Reply:
Also, the `$IsTar` check doesn't support `.tar.gz` or `.tar.xz`.

Reply:
`(strip_ext $Path) -match '\.tar$')` will fit all `.tar.xxx`, and `.tZ` is not supported by 7zip. But you regex is perfect :)

Reply:
Missed the `strip_ext` ... I take everything back 😄

Reply:
> Okay. But just using a line index isn't reliable.
> In [079c4a5](https://github.com/lukesampson/scoop/commit/079c4a57269a2a570e29d4326fcb615e98f0b0b4) it was `-3` now it's `-4` 😕

The `Invoke-ExternalCommand` write log to a file which has a blank end line...

So if you save the output to variable it will be the last 3 line while for file it is last 4 line.

Could you try `curl`'s installation? I've tested several tar, and [-4] is the right line number.

Reply:
Works fine 👍